axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/GrpcBackedResponseMessage.java

```diff
@@ -36,25 +40,41 @@
     private final QueryResponse queryResponse;
     private final Serializer messageSerializer;
     private final LazyDeserializingObject<R> serializedPayload;
+    private final Throwable exception;
     private final Supplier<MetaData> metadata;
 
     public GrpcBackedResponseMessage(QueryResponse queryResponse, Serializer messageSerializer) {
         this.queryResponse = queryResponse;
         this.messageSerializer = messageSerializer;
         this.metadata = new GrpcMetadata(queryResponse.getMetaDataMap(), messageSerializer);
-        if( queryResponse.hasPayload() && !"empty".equalsIgnoreCase(queryResponse.getPayload().getType())) {
+        if (queryResponse.hasMessage()) {
```

Comment:
So, `hasMessage()` means there is an exception? Shouldn't something like `isExceptional` make more sense on the `QueryResponse` class?

Reply:
Aaah of course, good point. Let's leave this for now. Agreeing with your judgement on this.